Add clarification about API resources to the Style Guide (#35119)
kubernetes/website: content/en/docs/contribute/style/style-guide.md

Set the value of the `replicas` field to 2. | Set the value of the `replicas` field to `2`.
{{< /table >}}

## Referring to Kubernetes API resources

This section talks about how we reference API resources in the documentation.

### Clarification about "resource"

> Kubernetes uses the word "resource" to refer to API resources, such as `pod`, `deployment`, and so on. We also use "resource" to talk about CPU and memory requests and limits.

Reviewer: I think it's worth adding one more line here directing a contributor to always refer to API resources as "API resources" to avoid confusion with CPU and memory resources

Author: Gotcha

### When to use Kubernetes API terminologies

The different Kubernetes API terminologies are:

- Resource type: the name used in the API URL (such as `pods`, `namespaces`)
- Resource: a single instance of a resource type (such as `pod`, `secret`)
- Object: a resource that serves as a "record of intent". An object is a desired state for a specific part of your cluster, which the Kubernetes control plane tries to maintain.

> Furthermore, you should always mention "resource" or "object" when referring to an API resource in docs. For example, use "a `Secret` object" over just "a `Secret`".

Reviewer: ```suggestion
Always use "resource" or "object" when referring to an API resource in docs. For example, use "a `Secret` object" over just "a `Secret`".
```

Author: Ah, yes, sorry for the wrong wordings

### API resource names

> Always format API resource names with use [UpperCamelCase](https://en.wikipedia.org/wiki/Camel_case), also known as PascalCase and code formatting.

Reviewer: ```suggestion
Always format API resource names using [UpperCamelCase](https://en.wikipedia.org/wiki/Camel_case), also known as PascalCase, and code formatting.
```

Author: thanks for spotting this grammatical problem

> For inline code in an HTML document, use the `<code>` tag. In a Markdown document, use the backtick (`` ` ``).

Reviewer: Does this render correctly, the backticks? I don't have access to the preview rn

Author: Yes it does, here's a screenshot
![image](https://user-images.githubusercontent.com/39672672/180618248-039b2cdb-b066-4ecd-90a3-959fde22056e.png)

> You may use the word "resource", "API", or "object" to clarify a Kubernetes resource type in a sentence.

Reviewer: I think delete this, since you say this in the preceding section

Author: Ah, yes

> Don't split an API object name into separate words. For example, use PodTemplateList, not Pod Template List.

Reviewer: ```suggestion
Don't split an API object name into separate words. For example, use `PodTemplateList`, not Pod Template List.
```

Author: Missed that, thanks for spotting!
